Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -5,120 +5,144 @@
  *
  */
 
-#include <GameplayEffectsNotificationBus.h>
+#include <Source/Components/Multiplayer/EnergyBallComponent.h>
+#include <Source/AutoGen/NetworkHealthComponent.AutoComponent.h>
+#include <Multiplayer/Components/NetworkTransformComponent.h>
+#include <Multiplayer/Components/NetworkRigidBodyComponent.h>
 #include <MultiplayerSampleTypes.h>
-#include <AzCore/Component/ComponentApplicationBus.h>
 #include <AzCore/Component/TransformBus.h>
-#include <AzCore/Settings/SettingsRegistry.h>
 #include <AzFramework/Physics/RigidBodyBus.h>
-#include <AzFramework/Physics/Collision/CollisionEvents.h>
-#include <AzFramework/Physics/SimulatedBodies/RigidBody.h>
-#include <Multiplayer/Components/NetworkTransformComponent.h>
-#include <Source/AutoGen/NetworkHealthComponent.AutoComponent.h>
-#include <Source/Components/Multiplayer/EnergyBallComponent.h>
-#include <Source/Components/Multiplayer/PlayerKnockbackEffectComponent.h>
 
 namespace MultiplayerSample
 {
+    AZ_CVAR(float, sv_EnergyBallImpulseScalar, 500.0f, nullptr, AZ::ConsoleFunctorFlags::Null, "A fudge factor for imparting impulses on rigid bodies due to weapon hits");
+
+    void EnergyBallComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<EnergyBallComponent, EnergyBallComponentBase>()
+                ->Version(1);
+        }
+        EnergyBallComponentBase::Reflect(context);
+    }
+
+    void EnergyBallComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        m_effect = GetExplosionEffect();
+        m_effect.Initialize();
+    }
+
+    void EnergyBallComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        ;
+    }
+
+#if AZ_TRAIT_CLIENT
+    void EnergyBallComponent::HandleRPC_BallExplosion([[maybe_unused]] AzNetworking::IConnection* invokingConnection, const AZ::Vector3& location)
+    {
+        AZ::Transform transform = AZ::Transform::CreateFromQuaternionAndTranslation(AZ::Quaternion::CreateIdentity(), location);
+        m_effect.TriggerEffect(transform);
+    }
+#endif
+
+
     EnergyBallComponentController::EnergyBallComponentController(EnergyBallComponent& parent)
         : EnergyBallComponentControllerBase(parent)
     {
+        ;
     }
 
     void EnergyBallComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
-        LoadEnergyBallSettings();
-
-        AzPhysics::RigidBody* body = nullptr;
-        Physics::RigidBodyRequestBus::EventResult(body, GetEntityId(), &Physics::RigidBodyRequests::GetRigidBody);
-        if (body)
-        {
-            body->RegisterOnCollisionBeginHandler(m_collisionHandler);
-        }
+#if AZ_TRAIT_SERVER
+        m_collisionCheckEvent.Enqueue(AZ::TimeMs{ 10 }, true);
```

Comment:
Why is 10 ms chosen here? It seems pretty arbitrary and disconnected from any other loop timing choices.

Reply:
Correct, completely arbitrary. I want it to tick every frame with best effort, but it's not a super critical system and so it's okay if the other 0 ms events starve this for a frame or two.

Reply:
i was more concerned about the opposite, which is that it will tick twice in the same frame every couple frames in the default case, which seems pretty wasteful, and we've got other cvars exposed that could accidentally cause this to be aligned either better _or_ worse. feels like at a minimum this should be exposed as a cvar too, but i'm not super concerned about this since it's just an energy ball, not a reusable system.